src/awkward/_v2/contents/content.py

```diff
@@ -449,6 +470,31 @@ def _carry_identifier(self, carry, exception):
         else:
             raise NotImplementedError
 
+    # def setidentifier(self, identifier):
+    #     if identifier is not None and len(self) != len(identifier):
+    #         raise TypeError(
+    #             "{0} content and its identifier must have the same length {1}".format(
+    #                 type(self).__name__, repr(identifier)
+    #             )
+    #         )
+    #     self._identifier = identifier
+
+    def setidentifier(self):
+        # FIXME
+        # nplike = self._nplike
+        # newidentifier = ak._v2.identifier.Identifier.empty(ak._v2.identifier.Identifier.newref(), ak._v2.identifier.Identifier.fieldloc(), len(self), 1, nplike)
+        # self._handle_error(
+        #         nplike[
+        #             "awkward_new_Identities",
+        #             newidentifier.dtype.type
+        #         ](
+        #             newidentifier.to(nplike),
+        #             len(self),
+        #         )
+        # )
+        # self.setidentities(newidentities)
+        return self
+
```

Comment:
There should not be a `setidentifier` interface because removing that is a long-standing issue: #176. Any tests that call `setidentifier` should just get removed. That way, the v1 → v2 transition would resolve that issue by simply not upgrading it.

Someday after that, I'll need to add a non-in-place `with_identifier`, but identifiers are a low priority now. (In the initial development, I thought they were going to be a lot more important.)